docs/src/userguide/language_basics.rst

```diff
@@ -890,6 +1280,11 @@ Conditional Compilation
 Some features are available for conditional compilation and compile-time
 constants within a Cython source file.
 
+.. note::
+
+    Cython currently does not support conditional compilation and compile-time
+    definitions in Pure Python mode.
```

Comment:
I propose split this to two notes to each subsection and add references to those issues: #4286, #4287